src/filters/body.rs

```diff
@@ -231,8 +231,8 @@ impl Decode for Json {
     const MIME: (mime::Name<'static>, mime::Name<'static>) = (mime::APPLICATION, mime::JSON);
     const WITH_NO_CONTENT_TYPE: bool = true;
 
-    fn decode<B: Buf, T: DeserializeOwned>(mut buf: B) -> Result<T, BoxError> {
-        serde_json::from_slice(&buf.to_bytes()).map_err(Into::into)
+    fn decode<B: Buf, T: DeserializeOwned>(buf: B) -> Result<T, BoxError> {
+        serde_json::from_slice(buf.chunk()).map_err(Into::into)
```

Comment:
`Buf::chunk` may not return the entire buffer (see https://docs.rs/bytes/1.0.0/bytes/trait.Buf.html#tymethod.chunk)

```suggestion
        serde_json::from_slice(&buf.copy_to_bytes(buf.remaining())).map_err(Into::into)
```